site-src/guides/implementers.md

```diff
@@ -159,16 +159,118 @@ the `metadata.generation` field of the object at the time the status is generate
 This allows users of the API to determine if the status is relevant to the current
 version of the object.
 
-
-### Resource details
+## TLS
+
+TLS is a large topic in Gateway API, with the set of capabilities continuing to
+expand. There is a [TLS guide](/guides/tls) that covers this topic in more depth
+from a user-facing perspective, but this section attempts to fill in some gaps
+from an implementer's perspective.
+
+### Listener Isolation
+Within Gateways, TLS config is currently tied exclusively to Listeners. To make
+that approach manageable, we're encouraging all implementations to work towards
+the goal of providing full "Listener Isolation" as defined below:
+
+Requests SHOULD match at most one Listener. For example, if Listeners are
+defined for "foo.example.com" and "*.example.com", a request to
+"foo.example.com" SHOULD only be routed using routes attached to the
+"foo.example.com" Listener (and not the "*.example.com" Listener).
+
+Implementations that do not support Listener Isolation MUST clearly document
+this. In the future, we plan to add Listener Isolation conformance tests to
+ensure this behavior is consistent across implementations that claim support for
+the feature.
```

Comment:
This text will likely bitrot, it might be helpful to link the relevant issues here, and then in the issues add a TODO list item to update the documentation here.

Reply:
Good call. Created https://github.com/kubernetes-sigs/gateway-api/issues/2803 and linked to it here.